TypeCobol.Analysis/Graph/BasicBlock.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace TypeCobol.Analysis.Graph
+{
+    /// <summary>
+    /// A Block of instruction of which the first instruction can be reached by an explicit control flow.
+    /// </summary>
+    /// <typeparam name="N">The Type Variable of an instruction</typeparam>
+    /// <typeparam name="D">The Type Variable for any extra Data information</typeparam>
+    public class BasicBlock<N, D> : ICloneable
+    {
+        /// <summary>
+        /// BasicBlock Index
+        /// </summary>
+        public int Index
+        {
+            get;
+            protected internal set;
+        }
+        /// <summary>
+        /// Instructions making up this block.
+        /// </summary>
+        public LinkedList<N> Instructions
+        {
+            get;
+            internal set;
+        }
+
+        /// <summary>
+        /// Data Information of this block.
+        /// </summary>
+        public D Data
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// The list of edge that leaves this block. The edges entries in the SuccessorEdges list of the ControlFlowGraph that contains this block.
+        /// </summary>
+        public List<int> SuccessorEdges
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// Flag on a basic block.
+        /// </summary>
+        [Flags]
+        public enum Flags : uint
+        {
+            Ending = 0x01 << 0,         //Flag if this basic block is an ending block.
+            Default = 0x01 << 1,        //Flag if this basic block is  default block for instance a WhenOther block.
+            Declaratives = 0x01 << 2,   //Flag if this basic block is inside a declaratives section.
+            Start = 0x01 << 3,          //Flag for a start node.
+            End = 0x01 << 4,            //Flag for a end node.
```

Comment:
Only tested for existence (!)

Reply:
Still to used, for instance the End flag is used to detect terminal block that contains a GOTO instruction. Default flag is used in whenCondBlock. No these flags are usefull.

Reply:
Ok but it has to be applied somewhere. How do you define an End block ? And how is it different from the `Ending` flag ?

Reply:
As discussed, the `End` flag is meant to identify a single end node for the whole graph. This is not implemented yet but an option could be added to build an additional node in the graph to which all `Ending` nodes would point.

Reply:
yes, for instance if we need to represent taht in an IDE.